lib/services/platform-service.ts

```diff
@@ -461,7 +461,9 @@ export class PlatformService implements IPlatformService {
 
 	public deployOnDevice(platform: string, buildConfig?: IBuildConfig): IFuture<void> {
 		return (() => {
-			this.installOnDevice(platform, buildConfig).wait();
+			if (!this.$options.debug) {
+				this.installOnDevice(platform, buildConfig).wait();
```

Comment:
Currently you have a method that is called deployOnDevice but in some special situations it doesn't deploy anything, it just starts the application. 
Since installOnDevice is public I would suggest you to change the deployOnDevice to be just startOnDevice (or some better name)  and call